### DISCUSSION TITLE
Highlight escape codes are not working correctly

### DISCUSSION THREAD

**Discussion by ljmf00** [community/community#14339] [Actions]
Currently, Github Actions UI can’t show highlighted escaped text correctly. See https://github.com/dlang/dmd/runs/5724204662?check_suite_focus=true#step:5:81 . You can compare with the raw log file and see the escaped sequence.

Running locally the escape shows something like this:
![image](https://user-images.githubusercontent.com/5050669/161813740-027bbaf8-b556-458a-8e7c-8df144a91b48.png)

Now it's looking like this:
![image](https://user-images.githubusercontent.com/5050669/161813770-66ed15af-c2ec-4267-9101-d52a268c435b.png)

**Comment by chrispat** [comment]
Other than the color differences, I am not seeing the differences in the screen shot.  Could you provide some additional annotation on your screen shots?

**Reply by ljmf00** [reply]
The background colour provided by the terminal escape code is not rendered on the GitHub output viewer, but it should render on a normal terminal.

![image](https://user-images.githubusercontent.com/5050669/161835669-badd1cf6-7eee-46b3-97d0-58e49a9656bd.png)
![image](https://user-images.githubusercontent.com/5050669/161835742-a9d883ec-86fb-40a4-a9dc-15bf7fefd544.png)

The second one is GitHub output viewer, and it doesn't render correctly. You can check the escape codes on the raw output hyperlink I provided in the initial thread description:

```
2022-03-28T16:52:40.9926397Z [1;31m-[m[1;31m    - uses: pre-commit/action@v2.0.3[m[7;31m [m
2022-03-28T16:52:40.9927102Z [1;32m+[m[1;32m    - uses: pre-commit/action@v2.0.3[m
```

**Reply by ethomson** [reply]
Thanks for the additional details! I'll make sure that this is on the backlog to investigate.
